test/unit/date/parse.js

```diff
@@ -471,6 +471,12 @@ QUnit.test( "should parse milliseconds in a day (A+)", function( assert ) {
 	assertParse( assert, "6330736900", "AAAAA", cldr, date1 );
 });
 
+QUnit.test( "should allow strict parsing of the date", function( assert ) {
+	date1 = new Date(2001,0,1);
+	assertParse( assert, "1/1/01", "M/d/yy", cldr, date1, true );
+	assertParse( assert, "1/1/2001", "M/d/yy", cldr, null, true );
+});
```

Comment:
well... 01 and 1 do not give incorrect value, where 2016 and 20 are incorrect values. this is what i was trying to avoid in first place.

I.e. people see format suggestion as `mm/dd/yy` and they might enter `1/12/16`, or  `01/12/16`, or `1/12/2016`. Only last one will produce incorrect Date value as a result of parsing.